### DISCUSSION TITLE
Transfer SPI NSS control from "radio" section

### DISCUSSION THREAD

**Discussion by IAmKapuze** [Lora-net/LoRaMac-node#1290] [Ideas]
SX1272/SX1276 has "Write/Read Buffer" code, which calls SpiInOut AND manually controls NSS. May be better way is controlling NSS into SpiInOut function?
I adapt SX1272/SX1276 code for using on a custom linux board with many /dev/spidevX.Y devices, and NSS control granted to the kernel. I call individual SPI transmitting (ioctl(..., SPI_IOC_MESSAGE(1), ...) in each SpiInOut calling. So I have to edit sx1272.c/sx1276.c to merge several calls SpiInOut to one, for holding NSS active while transmitting is active.

**Comment by mluis1** [comment]
Could you please take a look at [v5.0.0-branch](https://github.com/Lora-net/LoRaMac-node/tree/v5.0.0-branch)?
In this branch we have integrated the new Semtech radio drivers. These new drivers should address your concern.

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.

**Reply by IAmKapuze** [reply]
Yes, sorry for creating issue. I will consider creating discussion instead issue next time.
Yes, [radio driver](https://github.com/Lora-net/LoRaMac-node/tree/v5.0.0-branch/src/radio/sx127x_driver) from [v5.0.0 branch](https://github.com/Lora-net/LoRaMac-node/tree/v5.0.0-branch) looks more cleanly for realizing own devices. Creating user implemention of the HAL functions (including sx127x_hal_write/read) is a good idea. Thank you!
